compiler/rustc_middle/src/middle/resolve_lifetime.rs

```diff
@@ -64,7 +64,7 @@ pub struct ResolveLifetimes {
     /// Set of lifetime def ids that are late-bound; a region can
     /// be late-bound if (a) it does NOT appear in a where-clause and
     /// (b) it DOES appear in the arguments.
-    pub late_bound: FxHashMap<LocalDefId, FxHashSet<LocalDefId>>,
+    pub late_bound: FxHashMap<LocalDefId, FxIndexSet<LocalDefId>>,
```

Comment:
Is that value constructed in a deterministic order though? AFAICS no, `fn convert_named_region_map` uses the iteration order of `named_region_map.late_bound`, which is a `HirIdSet = FxHashSet`.

(The test case does repeatedly pass for me though...)